driver/kernel_queue.c

```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2023, Advanced Micro Devices, Inc.
+// SPDX-License-Identifier: MIT
+
+#include <stddef.h>
+#include <linux/device.h>
+#include <linux/export.h>
+#include <linux/err.h>
+#include <linux/fs.h>
+#include <linux/file.h>
+#include <linux/sched.h>
+#include <linux/slab.h>
+#include <linux/uaccess.h>
+#include <linux/compat.h>
+#include <linux/time.h>
+#include <linux/mm.h>
+#include <linux/mman.h>
+#include <linux/ptrace.h>
+#include <linux/dma-buf.h>
+#include <linux/fdtable.h>
+#include <linux/processor.h>
+#include <linux/pci.h>
+
+#include "chardev.h"
+#include "kernel_queue.h"
+#include "hsa_defs.h"
+
+/*  
+  Get the address to queue 0 and point the admin queue of the device to it.
+*/
+hsa_status_t air_queue_create(uint32_t size, uint32_t type, struct vck5000_device *dev) {
+
+  // Get the base address of the queue, subtract the physical offset, 
+  // and point our admin queue pointer to it
+  dev->admin_queue = dev->bram_bar + HERD_CONTROLLER_BASE_ADDR(dev->bram_bar, 0) - BRAM_PADDR;;
```

Comment:
too many ;;

Reply:
Addressed in [391351c](https://github.com/Xilinx/mlir-air/pull/250/commits/391351c6ed10c4cca742dad0a07c25f1bdb0666c)